### DISCUSSION TITLE
[Modding] Allow to define custom husk overlays for human husks

### DISCUSSION THREAD

**Discussion by TheNinjaScout** [FakeFishGames/Barotrauma#6768] [Unstable feedback]
At the moment, every single human husk (as in, every character that is spawned through a human being transformed by an alieninfection type affliction) will always have the same husk overlay on top of their face. 

Although it is possible to remove the "huskmask" texture (Human_husk_mask.png) from the human ragdoll, it won't prevent the overlay (Human_husk.png) defined as a wearable in the human ragdoll from being applied.

```xml
    <Wearable type="Husk">
      <sprite name="Husk" texture="Content/Characters/Human/Human_husk.png" hidewearablesoftype="Beard" sheetindex="0,0"/>
    </Wearable>
```
The above XML element can be changed only by overriding the human character, and even then it won't allow for each different human husk to have a different overlay texture.

```xml
    <huskmask texture="Content/Characters/Human/Human_husk_mask.png" />
```
The above XML element is found within the human ragdoll, but also works within a human husk ragdoll in case they have their own ragdoll. This texture seems to simply delete a portion of the face's texture, allowing the overlay to be applied easily to each different face.

I think that it should be made possible to define the overlay next to the "huskmask" texture within the ragdoll file itself, allowing different ragdolls to modify the textures differently.

**Comment by itchyOwl** [comment]
Thanks for the feedback. I agree on the main point. However, some clarifications:

Even thought the huskmask is defined in the human ragdoll, it's only applied for human husks. It's a cut off mask for the shader. So both editing the mask and removing it from being used would disable it in practice.

Previously we used sprite swapping. And this approach should still be supported. It's possible that it doesn't work right anymore due to all the code changes. If that's the case, it would be considered a bug that you should report and we should fix it. For modders, skipping the mask approach and using swappable heads would be easier, so I'd say we should support that approach.

From the top of my head, I think the mask will currently be enable for all the alien infections, when it should probably be only enabled for only the husk affliction.

In any case, the moddability aspects of these changes deserve a thought. Let's get back to this later!
